core-rust/state-manager/src/protocol/protocol_configs/config_printer.rs

```diff
@@ -0,0 +1,238 @@
+    );
+
+    if thresholds.is_empty() {
+        println!("{base_indent}    readiness_thresholds: vec![],");
+    } else {
+        println!("{base_indent}    readiness_thresholds: vec![");
+        for threshold in thresholds {
+            let ratio = threshold.required_ratio_of_stake_supported;
+            let epochs = threshold.required_consecutive_completed_epochs_of_support;
+            let est_duration = calculator.estimate_duration_of_full_epochs(epochs);
+            println!("{base_indent}        SignalledReadinessThreshold {{");
+            println!(
+                "{base_indent}            required_ratio_of_stake_supported: dec!({}),",
+                ratio
+            );
+            println!(
+                "{base_indent}            required_consecutive_completed_epochs_of_support: {}, // estimated: {}",
+                epochs,
+                display_duration(est_duration),
+            );
+            println!("{base_indent}        }},");
+        }
+        println!("{base_indent}    ],");
+    }
+    println!("{base_indent}}},");
+}
+
+#[test]
+fn print_readiness_signals() {
+    let base_indent = "        ";
+    for (network, config) in generate_network_configs() {
+        if !config.protocol_update_triggers.is_empty() {
+            println!("> Network: {network}");
+        }
+        for trigger in config.protocol_update_triggers {
+            if let ProtocolUpdateEnactmentCondition::EnactAtStartOfEpochIfValidatorsReady {
+                ..
+            } = trigger.enactment_condition
+            {
+                println!("{base_indent}    // =================================================================");
+                println!(
+                    "{base_indent}    // PROTOCOL_VERSION: \"{}\"",
+                    trigger.next_protocol_version
+                );
+                println!(
+                    "{base_indent}    // READINESS_SIGNAL: \"{}\"",
+                    trigger.readiness_signal_name()
+                );
+                println!("{base_indent}    // =================================================================");
+                println!();
+            }
+        }
+    }
+}
+
+fn display_duration(duration: Duration) -> String {
+    if duration < Duration::zero() {
+        panic!("Duration has to be >= 0")
+    }
+    let days = duration.num_days();
+    let hours = duration.num_hours() - 24 * duration.num_days();
+    let mins = duration.num_minutes() - 60 * duration.num_hours();
+    let secs = duration.num_seconds() - 60 * duration.num_minutes();
+    if days > 0 {
+        format!("{days} days, {hours} hours, {mins} mins, {secs} secs")
+    } else if hours > 0 {
+        format!("{hours} hours, {mins} mins, {secs} secs")
+    } else if mins > 0 {
+        format!("{mins} mins, {secs} secs")
+    } else if secs > 0 {
+        format!("{secs} secs")
+    } else {
+        "Immediate".to_string()
+    }
+}
```

Comment:
(optional) I've seen a few crates doing "human-readable duration".
I know it is 20 lines, but personally I'd accept an extra dev-only dependency (in exchange for avoiding a bug in a hand-crafted time math).